pypeit/coadd2d.py

```diff
@@ -1294,21 +1322,38 @@ def get_maskdef_dict(self, slit_idx, ref_trace_stack):
 
             # maskdef_slitcenters. This trace the slit center along the spectral direction.
             # But here we take only the value at the mid point
-            maskdef_slitcen_pixpos = self.stack_dict['slits_list'][0].maskdef_slitcen[self.nspec//2, slit_idx] + self.maskdef_offset
+
+            # TODO JFH now that we allow different image sizes with lists is this correct?
```

Comment:
@jhennawi I don't think this will work for images with different sizes, since `self.stack_dict['slits_list'].maskdef_slitcen` and `self.stack_dict['slits_list'].maskdef_objpos` depends on the slit tracing performed in each image (which will have different `MasterSlits`). I think `maskdef_slitcen_pixpos` and `imaskdef_objpos` will have to be lists/arrays and  `imaskdef_slitcen_dspat` and `imaskdef_objpos_dspat` cannot be the mean over every `iexp`, but they will have to be lists/arrays with one value for each image. 
I'll think how we can fix this.
However, I believe if coadd2d is run for images with the same `MasterSlits` it should still work the same.

Reply:
@jhennawi I tried running `coadd2d` for images with the **same size** and these changes do not work either. The script crashes. We need to fix it.

Reply:
I fixed these issues. I agree that different sizes requires some thought though. For now, things seem to work on the tests. Please also have a look.